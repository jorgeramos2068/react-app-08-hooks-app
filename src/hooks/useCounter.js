import {useState} from 'react';

const useCounter = (initialValue = 10) => {
  const [counter, setCounter] = useState(initialValue);

  const increment = (factor = 1) => {
    setCounter(counter + factor);
  };

  const decrement = (factor = 1) => {
    setCounter(counter - factor);
  };

  const reset = () => {
    setCounter(initialValue);
  }

  return {
    counter,
    increment,
    decrement,
    reset
  };
};

export default useCounter;
