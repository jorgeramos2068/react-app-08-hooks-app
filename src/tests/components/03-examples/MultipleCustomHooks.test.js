import React from 'react';
import {shallow} from 'enzyme';
import MultipleCustomHooks from '../../../components/03-examples/MultipleCustomHooks';
import useFetch from '../../../hooks/useFetch';
import useCounter from '../../../hooks/useCounter';
jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Tests for <MultipleCustomHooks />', () => {
  useCounter.mockReturnValue({
    counter: 10,
    increment: () => {}
  });

  test('Should display <MultipleCustomHooks /> correctly', () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should show remote data', () => {
    useFetch.mockReturnValue({
      data: [{
        author: 'Test author',
        quote: 'Test quote'
      }],
      loading: false,
      error: null
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper.find('.alert').exists()).toBe(false);
    expect(wrapper.find('.mb-0').text().trim()).toBe('Test quote');
    expect(wrapper.find('footer').text().trim()).toBe('Test author');
  });
});
