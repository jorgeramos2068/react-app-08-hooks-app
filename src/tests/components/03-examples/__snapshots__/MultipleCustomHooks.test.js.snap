// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <MultipleCustomHooks /> Should display <MultipleCustomHooks /> correctly 1`] = `
<div>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next couter
  </button>
</div>
`;
